nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessGroupResource.java

```diff
@@ -4173,6 +4187,205 @@ private void sanitizeRegistryInfo(final VersionedProcessGroup versionedProcessGr
+                    value = "The process group name.",
+                    required = true
+            )
+            @FormDataParam("groupName") final String groupName,
+            @ApiParam(
+                    value = "The process group X position.",
+                    required = true
+            )
+            @FormDataParam("position-x") final Double positionX,
+            @ApiParam(
+                    value = "The process group Y position.",
+                    required = true
+            )
+            @FormDataParam("position-y") final Double positionY,
+            @ApiParam(
+                    value = "The client id.",
+                    required = true
+            )
+            @FormDataParam("clientId") final String clientId,
+            @ApiParam(
+                    value = "Acknowledges that this node is disconnected to allow for mutable requests to proceed.",
+                    required = false
+            )
+            @FormDataParam(DISCONNECTED_NODE_ACKNOWLEDGED) @DefaultValue("false") final Boolean disconnectedNodeAcknowledged,
+            @FormDataParam("file") final InputStream in) throws IOException {
+
+        // ensure the group name is specified
+        if (StringUtils.isBlank(groupName)) {
+            throw new IllegalArgumentException("The process group name is required.");
+        }
+
+        if (StringUtils.isBlank(groupId)) {
+            throw new IllegalArgumentException("The parent process group id is required");
+        }
+
+        if (positionX == null) {
+            throw new IllegalArgumentException("The x coordinate of the proposed position must be specified.");
+        }
+
+        if (positionY == null) {
+            throw new IllegalArgumentException("The y coordinate of the proposed position must be specified.");
+        }
+
+        if (StringUtils.isBlank(clientId)) {
+            throw new IllegalArgumentException("The client id must be specified");
+        }
+
+        // get the contents of the InputStream as a String
+        String stringContent;
+        if (in != null) {
+            try {
+                stringContent = IOUtils.toString(in, StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new IOException("Unable to read the InputStream", e);
+            }
+        } else {
+            logger.warn("The InputStream is null");
+            throw new NullPointerException("The InputStream is null");
+        }
+
+        // deserialize content to a VersionedFlowSnapshot
+        VersionedFlowSnapshot deserializedSnapshot;
+
+        if (stringContent.length() > 0) {
+            try {
+                deserializedSnapshot = MAPPER.readValue(stringContent, VersionedFlowSnapshot.class);
+            } catch (JsonParseException jpe) {
+                logger.warn("Parsing uploaded JSON failed", jpe);
+                return Response.status(Response.Status.OK).entity(INVALID_JSON_RESPONSE).type("application/json").build();
```

Comment:
The call to `type()` indicates that the string provided to `entity()` should be formatted as JSON, which corresponds to the `@Produces(MediaType.APPLICATION_JSON)` annotation on this method.

On closer review, it looks like the status should be `Response.Status.BAD_REQUEST` to return an HTTP 400 instead of `Response.Status.OK`, otherwise wouldn't the JavaScript handling call success in this case?

Stepping back for a moment, is there a reason for returning a response at this point instead of throwing an exception and having the framework return an appropriately formatted response?